src/ConfigAggregator.php

```diff
@@ -108,13 +108,7 @@ private function removeDuplicateProviders(iterable $providers): iterable
         $uniqueProviders = [];
 
         foreach ($providers as $provider) {
-            if (in_array($provider, $uniqueProviders)) {
-                if (is_object($provider)) {
-                    $provider = $provider::class;
-                }
-                if (! is_string($provider)) {
-                    $provider = serialize($provider);
-                }
+            if (get_debug_type($provider) === 'string' && in_array($provider, $uniqueProviders, true)) {
```

Comment:
`get_debug_type` will be always string.

```suggestion
            if (!is_string($provider) && !is_object($provider)) {
                continue;
            }
            
            if (in_array($provider, $uniqueProviders, true)) {
```

Reply:
Always string?
For instances of an anonymous class it returns `class@anonymous`
For instances of a real class it returns the class name
For anonymous functions and generators it returns `Closure`
For arrays it returns `array`
For class-strings it returns `string`

Reply:
`get_debug_type` is used for printing debug info. If you want to check for string use `is_string()`

Reply:
@alexmerlin `get_debug_type` suggested in another comment was for reporting the problematic provider not for conditionals.

For example, when `$provider` can be something other than string we would report it this way:

```php
        return new self(sprintf(
            '%s is registered more than once. Config providers should be unique. In case a specific order is'
            . ' required, please double check before deleting the duplicate(s).',
            is_string($provider) ? $provider : get_debug_type($provider)
        ));
```